src/routes/safe/components/Apps/components/SecurityFeedbackModal/Slider.tsx

```diff
@@ -0,0 +1,207 @@
+import React from 'react'
+import styled from 'styled-components'
+import { useState, useRef, useEffect } from 'react'
+import { Button } from '@gnosis.pm/safe-react-components'
+import { Box } from '@material-ui/core'
+import { ThemeColors } from '@gnosis.pm/safe-react-components/dist/theme'
+
+type SliderProps = {
+  onCancel: () => void
+  onComplete: () => void
+}
+
+type SliderState = {
+  translate: number
+  activeSlide: number
+  transition: number
+  renderedSlides: React.ReactElement[]
+}
+
+const SLIDER_TIMEOUT = 500
+
+const Slider: React.FC<SliderProps> = ({ onCancel, onComplete, children }) => {
+  const allSlides = React.Children.toArray(children).filter(Boolean) as React.ReactElement[]
+  const stateRef = useRef<SliderState>({
+    translate: 1,
+    activeSlide: 0,
+    transition: 0.45,
+    renderedSlides: [],
+  })
+  const firstSlide = allSlides?.[0]
+  const secondSlide = allSlides?.[1]
+  const lastSlide = allSlides?.[allSlides?.length - 1]
+  const [disabledBtn, setDisabledBtn] = useState(false)
+  const [state, setState] = useState<SliderState>({
+    translate: 1,
+    activeSlide: 0,
+    transition: 0.45,
+    renderedSlides: [lastSlide, firstSlide, secondSlide],
+  })
+
+  stateRef.current = state
+
+  useEffect(() => {
+    const smooth = (e: TransitionEvent) => {
+      if ((e.target as HTMLElement).className.includes('inner')) {
+        smoothTransition()
+      }
+    }
+
+    const transitionEnd: any = document.addEventListener('transitionend', smooth)
```

Comment:
Probably because i set it quickly for continue with another thing as the type i used was giving me trouble. Then i forgot to address it again 😁. I'll fix it!

Reply:
Fixed. In fact i was removing air (undefined) 😬. So now I'm removing the real event